packaging/installer/kickstart.sh

```diff
@@ -2089,6 +2124,31 @@ validate_args() {
 
     SELECTED_RELEASE_CHANNEL="${RELEASE_CHANNEL}"
   fi
+
+  if [ -n "${INSTALL_MAJOR_VERSION}" ] && [ -n "${INSTALL_VERSION}" ]; then
+    fatal "Only one of --install-version or --install-major-version may be specified." F0515
+  fi
+
+  handle_major_version # Appropriately updates INSTALL_VERSION if INSTALL_MAJOR_VERSION is set.
+
+  if [ -n "${NETDATA_OFFLINE_INSTALL_SOURCE}" ] && [ -n "${INSTALL_VERSION}" ]; then
+      fatal "Specifying an install version alongside an offline install source is not supported." F050A
+  fi
+
+  if [ "${NETDATA_AUTO_UPDATES}" = "default" ]; then
+    if [ -n "${NETDATA_OFFLINE_INSTALL_SOURCE}" ] || [ -n "${INSTALL_VERSION}" ]; then
+      AUTO_UPDATE=0
+    else
+      AUTO_UPDATE=1
+    fi
+  elif [ "${NETDATA_INSTALL_MAJOR_VERSION}" ]; then
+    warning "Forcibly disabling auto updates as a specific major version was requested."
+    AUTO_UPDATE=0
+  elif [ "${NETDATA_AUTO_UPDATES}" = 1 ]; then
+    AUTO_UPDATE=1
+  else
+    AUTO_UPDATE=0
```

Comment:
When the user installs the latest major available for any channel we need to enforce our default behaviour; autoupdate, this needs some extra work and somehow make it clear to the user that from now on you will receive the latest version ever for this particular channel. Now the question is how you 'make sure that the major version you are installing is the latest, maybe a new endpoint under `https://repo.netdata.cloud/releases/latest/nightly|stable` which will return vX

Reply:
> When the user installs the latest major available for any channel we need to enforce our default behaviour; autoupdate

There is currently no support in the updater code for this, and I see little value because users who go out of their way to use this option are all but certain to be managing updates on their systems themselves.

> Now the question is how you 'make sure that the major version you are installing is the latest, maybe a new endpoint under https://repo.netdata.cloud/releases/latest/nightly|stable which will return vX

You do this by not specifying the `--install-major-version` option.

The intent here is not to change default behavior. Run without the new `--install-major-version` option, the kickstart script should work _identically_ to how it does currently. The goal is to provide a shortcut for users who are choosing to stick with an _old_ major version for compatibility reasons so that they don’t have to look up the correct value to specify with the existing `--install-version` option.